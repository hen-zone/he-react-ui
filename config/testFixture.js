import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
