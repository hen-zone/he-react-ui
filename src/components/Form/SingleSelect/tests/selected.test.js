import fixture from '../fixtures/selected.fixture.js'
import testFixture from '../../../../../config/testFixture'

testFixture(fixture, 'selected')
