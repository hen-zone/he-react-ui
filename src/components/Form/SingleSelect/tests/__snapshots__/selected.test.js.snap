// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnClickOutside(SingleSelect) /> rendered correctly with selected fixture 1`] = `
<OnClickOutside(SingleSelect)
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  id="demo"
  name="demo"
  options={
    Array [
      Object {
        "label": "Option 1",
        "value": "1",
      },
      Object {
        "label": "Option 2",
        "value": "2",
      },
      Object {
        "label": "Option 3",
        "value": "3",
      },
    ]
  }
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
  value="3"
>
  <SingleSelect
    disableOnClickOutside={[Function]}
    disabled={false}
    enableOnClickOutside={[Function]}
    error=""
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    id="demo"
    inline={false}
    name="demo"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Option 1",
          "value": "1",
        },
        Object {
          "label": "Option 2",
          "value": "2",
        },
        Object {
          "label": "Option 3",
          "value": "3",
        },
      ]
    }
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    value="3"
  >
    <div
      className="outer"
      options={
        Array [
          Object {
            "label": "Option 1",
            "value": "1",
          },
          Object {
            "label": "Option 2",
            "value": "2",
          },
          Object {
            "label": "Option 3",
            "value": "3",
          },
        ]
      }
    >
      <div
        className="select"
        id="demo"
        onClick={[Function]}
      >
        <span>
          Option 3
        </span>
        <Icon
          className="caret"
          name="DropDown"
        >
          <DropDown
            className="icon defaultSize caret"
          >
            <svg
              className="icon defaultSize caret"
              version="1.1"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                transform="translate(15.000000, 7.000000) rotate(90.000000)"
              >
                <path
                  d="M2,6.93247919 L2,-0.932479194 C2,-1.20862157 2.22385763,-1.43247919 2.5,-1.43247919 C2.61696825,-1.43247919 2.7302347,-1.39147108 2.8200922,-1.31658983 L7.53906723,2.61588936 C7.75120568,2.7926714 7.77986771,3.10795375 7.60308567,3.3200922 C7.58372606,3.34332373 7.56229876,3.36475103 7.53906723,3.38411064 L2.8200922,7.31658983 C2.60795375,7.49337187 2.2926714,7.46470984 2.11588936,7.25257139 C2.04100811,7.16271389 2,7.04944745 2,6.93247919 Z"
                />
              </g>
            </svg>
          </DropDown>
        </Icon>
      </div>
      <ul
        className="options"
      >
        <li
          className="option"
          key="1"
          onClick={[Function]}
        >
          Option 1
        </li>
        <li
          className="option"
          key="2"
          onClick={[Function]}
        >
          Option 2
        </li>
        <li
          className="option selected"
          key="3"
          onClick={[Function]}
        >
          Option 3
        </li>
      </ul>
    </div>
  </SingleSelect>
</OnClickOutside(SingleSelect)>
`;
